File: src/main/java/pl/poul12/matchzone/model/Appearance.java
package pl.poul12.matchzone.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import pl.poul12.matchzone.model.enums.Eyes;
import pl.poul12.matchzone.model.enums.HairColour;
import pl.poul12.matchzone.model.enums.Physique;

import javax.persistence.*;

@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
@Entity
public class Appearance {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Eyes eyes;
    private HairColour hairColour;
    private Short height;
    private Physique physique;
    private String about;
    private String hobbies;
    @JsonBackReference
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    @EqualsAndHashCode.Exclude
    User user;
}

File: match-zone-app/src/app/appearance.ts
import {Eyes} from "../eyes";
import {HairColour} from "../hair-colour";
import {Physique} from "./physique";
import {User} from "./user";

export class Appearance{
    id: number;
    eyes: Eyes;
    hairColour: HairColour;
    height: number;
    physique: Physique;
    about: string;
    hobbies: string;
    user: User;
}
